### DISCUSSION TITLE
Is cargo check necessary in buildDepsOnly?

### DISCUSSION THREAD

**Discussion by szlend** [ipetkov/crane#476] [Q&A] [answered]
I've been wondering why `cargo check` is necessary in `buildDepsOnly`. It adds quite a bit to the compilation time as you're basically compiling the deps twice (`cargo check` + `cargo build`).

There's a comment in `buildDepsOnly.nix` saying:
```
# First we run `cargo check` to cache cargo's internal artifacts, fingerprints, etc. for all deps.
# Then we run `cargo build` to actually compile the deps and cache the results
```

However I tried disabling the cargo check step and the dep caching still appears to work fine and I didn't notice any dependencies being recompiled in the `buildPackage` step. Is it safe to disable this or will I eventually run into problems? I think it's worth leaving a comment with some more context, like specific instances where this is necessary.

**Comment by dpc** [comment, marked as answer]
Might be wrong, but the "check" part might be needed for clippy and `cargo doc`. So if someone doesn't need them they might want to skip `cargo check` on deps.

**Reply by szlend** [reply]
I can confirm that it's definitely necessary for caching `cargo clippy`. We have separate debug/release builds, with `debug` builds being used for checks so I think it should be fine to disable this for our `release` builds.
